neqo-http3/src/connection_client.rs

```diff
@@ -426,30 +434,32 @@ impl Http3Client {
                 t.reset_receiving_side();
             }
             if t.done() {
-                self.base_handler.transactions.remove(&stop_stream_id);
+                self.base_handler
+                    .transactions
+                    .remove(&stop_stream_id.as_u64());
             }
         }
         Ok(())
     }
 
-    fn handle_goaway(&mut self, goaway_stream_id: u64) -> Res<()> {
+    fn handle_goaway(&mut self, goaway_stream_id: StreamId) -> Res<()> {
         qinfo!([self], "handle_goaway");
         // Issue reset events for streams >= goaway stream id
         for id in self
             .base_handler
             .transactions
             .iter()
-            .filter(|(id, _)| **id >= goaway_stream_id)
+            .filter(|(id, _)| **id >= goaway_stream_id.as_u64())
```

Comment:
This is where you have to consider implementing `PartialCmp` for `StreamId`.  I think that's not a function you want to have implemented on the struct, so you would need to unbox in this function.